### DISCUSSION TITLE
How to deal with nullable fields?

### DISCUSSION THREAD

**Discussion by fairking** [go-rel/rel#350] [Q&A]
Sorry, searched docs, nothing around this topic. All [examples](https://github.com/go-rel/gin-example) and [documentation](https://go-rel.github.io/basics/) has no nullable/optional fields.

So the question is how to use migrations with nullable columns (how to describe nullable columns) and how to define them as entity fields (optional fields).

``` go
type Todo struct {
    ID   int
    Active bool // ???
}
```

``` go
schema.CreateTable("todos", func(t *rel.Table) {
        t.ID("id")
        t.Bool("active") // ???
    })
```

**Comment by Fs02** [comment]
you can use `rel.Required(false)`

https://github.com/go-rel/sql/blob/be5e6ac51f8646ac1d56968ff7bf91c764ff46c5/specs/migration.go#L22

**Reply by fairking** [reply]
I don't understand. Your example shows `required('true')`. What sense does it make? I have the same result without `required("true")`.

**Reply by fairking** [reply]
So you are assuming the default is `required('false')`. Then how can I handle `null` in `go` having such situation?

**Reply by Fs02** [reply]
> So you are assuming the default is required('false').

yes, it's the default

> Then how can I handle null in go having such situation?

you can just use pointer, example:

```go
type Todo struct {
    ID   int
    Active *bool // <= use pointer
}
```
